From 605d6dab5106587a59aaaa241cc1089a310704f0 Mon Sep 17 00:00:00 2001
Message-Id: <605d6dab5106587a59aaaa241cc1089a310704f0.1547150348.git.sandeep.sheriker@microchip.com>
From: Sandeep Sheriker M <sandeepsheriker.mallikarjun@microchip.com>
Date: Mon, 7 Jan 2019 13:47:02 -0700
Subject: [EGT][patch v5 1/3] egt:configure.ac: Add check for libmagic

add check for libmagic library and this is required for querying
mimetype of a file.

Signed-off-by: Sandeep Sheriker M <sandeep.sheriker@microchip.com>
---
 configure.ac | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/configure.ac b/configure.ac
index aab15fa..ef2d58b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -163,6 +163,15 @@ if test "x${have_libjpeg}" = xyes; then
    AC_DEFINE(HAVE_LIBJPEG, 1, [Have libjpeg support])
 fi
 
+AC_SEARCH_LIBS([magic_open], [magic], [MAGIC_LIB=-lmagic], [
+  AC_MSG_ERROR([required library libmagic not found])
+])
+
+AC_CHECK_HEADER([magic.h],
+  [AC_DEFINE(HAVE_LIBMAGIC, 1, [Have libmagic support])],
+  [AC_MSG_ERROR([required header magic.h not found])
+])
+
 AC_LANG_PUSH([C++])
 AC_CHECK_HEADERS([rapidxml/rapidxml.hpp], [have_rapidxml=yes], [have_rapidxml=no])
 AM_CONDITIONAL([HAVE_RAPIDXML_RAPIDXML_HPP], [test "x${have_rapidxml}" = xyes])
-- 
2.7.4

